datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -203,7 +201,7 @@ fn split_eq_and_noneq_join_predicate_with_nulls_equality(
 
 async fn union_rels(
     rels: &[Rel],
-    ctx: &SessionContext,
+    ctx: &SessionState,
```

Comment:
In the original issue I also mention that we could declare a trait with just the method used by the substrait producer / consumer so users could provide their own shim.